packages/spektate/src/pipeline/AzureDevOpsPipeline.ts

```diff
@@ -130,37 +142,48 @@ export class AzureDevOpsPipeline implements IPipeline {
     if (releaseIds && releaseIds!.size === 0) {
       return this.releases;
     }
-    const json = await HttpHelper.httpGet<any>(
-      this.getReleaseUrl(releaseIds),
-      this.pipelineAccessToken
-    );
-
-    const releases: IRelease[] = [];
-    for (const row of json.data.value) {
-      const release: IRelease = {
-        URL: row.release._links.web.href,
-        finishTime: new Date(row.completedOn),
-        id: row.release.id,
-        lastUpdateTime: new Date(row.lastModifiedOn),
-        queueTime: new Date(row.queuedOn),
-        releaseName: row.release.name,
-        startTime: new Date(row.startedOn),
-        status: row.deploymentStatus
-      };
-
-      if (row.release.artifacts.length > 0) {
-        const defRef = row.release.artifacts[0].definitionReference;
-        release.imageVersion = defRef.version.id;
-
-        if (defRef.registryurl) {
-          release.registryURL = defRef.registryurl.id;
-        }
-        if (defRef.resourcegroup) {
-          release.registryResourceGroup = defRef.resourcegroup.id;
+
+    try {
+      const json = await HttpHelper.httpGet<any>(
+        this.getReleaseUrl(releaseIds),
+        this.pipelineAccessToken
+      );
+
+      if (!json.data || !json.data.value) {
+        throw new Error(
+          "Data could not be fetched from Azure DevOps. Please check the personal access token, org and project name."
+        );
+      }
+
+      const releases: IRelease[] = [];
```

Comment:
Is there a reason to track them in this array? seems like the effect of this loops is to push to `this.releases`